SearchImplem/LocalSearchAutoComplete.swift

```diff
@@ -22,57 +22,76 @@ class LocalSearchAutocomplete: UIViewController, UITableViewDataSource, UITableV
         return completer
     }()
     
-    var searchResults = [MKLocalSearchCompletion]()
-    
+    var predictions = [City]()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         searchCompleter.delegate = self
     }
     
     func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        searchResults = []
+        defer { tableView.reloadData() }
+        predictions = []
         guard searchText != "" else {
-            tableView.reloadData()
             return
         }
         searchCompleter.queryFragment = searchText
-        tableView.reloadData()
     }
     
     func completerDidUpdateResults(_ completer: MKLocalSearchCompleter) {
         for item in completer.results {
             if item.title.contains(",") {
```

Comment:
Why do you check for a comma here?

Reply:
It's the workaround to remove unwanted values.